app/lib/params_validator.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+AVAILABLE_STOCKS = %w[GPW NC].freeze
+AVAILABLE_CONDITIONS = %w[company_active].freeze
+
+class ParamsValidator
```

Comment:
I think the name is to generic. This class handles params in specific case, not all params in the app.

You could do something like this:

```suggestion
module Companies
  class IndexParamsValidator
```

or something similar.

Reply:
Also, personally, I like to keep things directly related to web layer a little bit separated. This is a validator of web request input before passing it into your app layer (service objects, etc.). This is similar to presenting the output from app layer (service objects, etc) before passing it into view to generated a HTML template (web layer).

I would keep such validators in a place similar to the place where I keep presenters. In your case it would be probably `app/requests/` or `app/request_params/`, etc. For example: `app/requests/companies/index_params_validator.rb` or `app/requests/companies_params_validator.rb`.

Reply:
Note: of course you could argue that the boundaries between the layers are in a little different places (maybe someone could say the params validation should inside the service, etc.).

Reply:
Yes there could be many different ways to designate boundaries between the layers, also people may look at this in many different aspects. It was little bit strange for me on the begin to put this into the `requests` , but I'm learning and I'm little bit :) backward with knowledge than you, so I look at this and I'm finding a way of your thinking, because I saw many times that it was correct. So I do  `app/requests/companies/index_params_validator.rb` . I'm not saying that I will not change something when I got some more knowledge. In this case I think it will clear when I have  more files to categorize.